### PR TITLE
perf(permission): add row and column cache to optimize performance

### PR DESCRIPTION
**Pull request by ybzky**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets-ui/src/controllers/permission/sheet-permission-interceptor-canvas-render.controller.ts [hunk 1]
```diff
@@ -72,26 +72,16 @@ export class SheetPermissionInterceptorCanvasRenderController extends RxDisposab
...
-
-                    const haveNotPermission = protectionLapRange.some((range) => {
-                        const { startRow, startColumn, endRow, endColumn } = range;
-                        for (let row = startRow; row <= endRow; row++) {
-                            for (let col = startColumn; col <= endColumn; col++) {
-                                const permission = (worksheet.getCell(row, col) as (ICellDataForSheetInterceptor & { selectionProtection: ICellPermission[] }))?.selectionProtection?.[0];
-                                if (permission?.[UnitAction.Edit] === false) {
-                                    return true;
-                                }
+                    const permissionIds: string[] = [];
```
Comment (wzhudev) [inline packages/sheets-ui/src/controllers/permission/sheet-permission-interceptor-canvas-render.controller.ts:75]: Could this be done in a single loop? 

Supposing that the first protection rule forbids the command, the current code would still loop through all rules and ranges. But if you do it in a single loop, you can skip all other rules.

File: packages/sheets-ui/src/controllers/permission/sheet-permission-interceptor-canvas-render.controller.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ export class SheetPermissionInterceptorCanvasRenderController extends RxDisposab
                     if (!target) {
                         return false;
                     }
-                    const { worksheet, unitId, subUnitId } = target;
+                    const { unitId, subUnitId } = target;
```
Comment (wzhudev) [inline packages/sheets-ui/src/controllers/permission/sheet-permission-interceptor-canvas-render.controller.ts:64]: Remove L26 dead code.

**Comment by wzhudev** [comment]
The title of this pull request shall be changed to better summarize what have been changed.

**Review by VicKun4937 — APPROVED** [review]

**Pull request merged**